File: android-v0-compose/app/src/main/java/com/xlwl/AiMian/data/model/OssModels.kt
package com.xlwl.AiMian.data.model

data class OssConfig(
  val endpoint: String,
  val bucketName: String,
  val region: String,
  val cdnDomain: String? = null
)

data class OssUploadResult(
  val objectKey: String? = null,
  val url: String? = null
)

data class OssUploadCompleteRequest(
  val sessionId: String,
  val questionIndex: Int,
  val ossUrl: String,
  val cdnUrl: String? = null,
  val fileSize: Long? = null,
  val duration: Long? = null
)

File: iOS-v0/Sources/Models/OssModels.swift
import Foundation

/// OSS 上传配置与上传结果 - 对齐 Android OssModels
struct OssConfig: Decodable, Sendable {
  let endpoint: String
  let bucketName: String
  let region: String
  let cdnDomain: String?
}

struct OssUploadResult: Decodable, Sendable {
  let objectKey: String?
  let url: String?
}

/// 通知后端 OSS 上传完成
struct OssUploadCompleteRequest: Encodable, Sendable {
  let sessionId: String
  let questionIndex: Int
  let ossUrl: String
  let cdnUrl: String?
  let fileSize: Int64?
  let duration: Int64?
}

/// 绑定实时面试某一沟通轮次对应的答题视频 - 对齐 Android AttachConversationTurnVideoBody
struct AttachConversationTurnVideoBody: Encodable, Sendable {
  let videoUrl: String?
  let videoPath: String?
  let durationMs: Int?
}

struct AttachConversationTurnVideoData: Decodable, Sendable {
  let videoUrl: String?
}
